### PR TITLE
Snyk_workflow add

### PR DESCRIPTION
**Pull request by SadFaceSmith**
What this PR does / why we need it:
This PR adds a short GH Action to generate and upload SBOM reports via Snyk.

Which issue(s) this PR fixes:

Fixes #

Special notes for your reviewer:

FYSA @grafana/security-team

### CODE REVIEW

**Comment by pracucci** [comment]
@SadFaceSmith Thanks for opening this PR. Could you check why the new CI job is failing, please?

**Comment by SadFaceSmith** [comment]
@pracucci Apologies, I had a leftover Node.js step in there. It should be good to go now.

File: .github/workflows/snyk.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Snyk SBOM CI
```
Comment (56quarters) [inline .github/workflows/snyk.yml:1]: nit: could you please rename this to match the other CI steps that get run, e.g. `snyk-sbom-ci`

**Review by 56quarters — CHANGES_REQUESTED** [review]
It's hard to tell what this CI step is meant to accomplish on its own. Could you add some documentation to the [`contribution` docs](https://github.com/grafana/mimir/tree/main/docs/internal/contributing) that explains what snyk is doing, how contributors should deal with any warnings or errors raised, etc.

**Pull request closed without merging**